feat(curriculum): add css-flexbox-quiz

henrysmile2071 opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #56187 

<!-- Feel free to add any additional description of changes below this line -->
1. Ran locally passing: ```FCC_CHALLENGE_ID=66ed8fe7f45ce3ece4053eb2 pnpm run test:curriculum```

henrysmile2071 committed:
feat: flexbox quiz markdown

henrysmile2071 committed:
Merge branch 'freeCodeCamp:main' into feat/add-css-flexbox-quiz

henrysmile2071 committed:
fix: format challenge text

gikf commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
````diff
@@ -17,439 +17,524 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 
-Placeholder distractor 1
+```css
+
+div { 
+  flex: 1; 
+}
+
````
```suggestion
div { 
  flex: 1; 
}
```
No need for additional blank lines here. It applies to other code blocks as well.

henrysmile2071 replied:
Ok, I've removed the additional blank lines for all the code blocks

henrysmile2071 committed:
fix: remove additional blank lines for code blocks

a2937 commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --text--
 
-Placeholder question
+What css property is set to enable the flexbox layout for the <div> element?  
```
I'm fairly sure that we want HTML elements like div to have backticks around them like this: `div`.

henrysmile2071 replied:
Oh, thanks for the catch, I have added the backticks.

henrysmile2071 committed:
fix: add backticks for elements in question

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 ---
 
-Placeholder distractor 2
+Items are aligned such that their flex container baselines align
```
```suggestion
Items are aligned such that their flex container baselines align.
```

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --answer--
 
-Placeholder answer
+Auto-size items to be equally large to fill the container
```
```suggestion
Auto-size items to be equally large to fill the container.
```

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --text--
 
-Placeholder question
+Items are evenly distributed along the main axis. The spacing between each pair of adjacent items is the same. The items on the sides are flush againt the main axis edge. Which value in `justify-content` matches this description?
```
```suggestion
Items are evenly distributed along the main axis. The spacing between each pair of adjacent items is the same. The items on the sides are flush againt the main axis edge.

Which value in `justify-content` matches this description?
```

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --text--
 
-Placeholder question
+Items are evenly distributed along the main axis. The spacing between each pair of adjacent items and against the main axis edge are the same. Which value in `justify-content` matches this description?
```
```suggestion
Items are evenly distributed along the main axis. The spacing between each pair of adjacent items and against the main axis edge are the same.

Which value in `justify-content` matches this description?
```

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --text--
 
-Placeholder question
+Items are evenly distributed along the main axis. The spacing between each pair of adjacent items is the same. The items on the sides have half the spacing against the main axis edge. Which value in `justify-content` matches this description?
```
```suggestion
Items are evenly distributed along the main axis. The spacing between each pair of adjacent items is the same. The items on the sides have half the spacing against the main axis edge.

Which value in `justify-content` matches this description?
```

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --text--
 
-Placeholder question
+Items are flush against the cross-axis start. Which value in `align-items` matches this description?
```
```suggestion
Items are flush against the cross-axis start.

Which value in `align-items` matches this description?
```

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --text--
 
-Placeholder question
+Items are flush against the cross-axis end. Which value in `align-items` matches this description?
```
```suggestion
Items are flush against the cross-axis end.

Which value in `align-items` matches this description?
```

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --text--
 
-Placeholder question
+Items are aligned along the main axis from left to right. Which value in `flex-direction` matches this description?
```
```suggestion
Items are aligned along the main axis from left to right.

Which value in `flex-direction` matches this description?
```

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --distractors--
 
-Placeholder distractor 1
+Centering an item in a container
```
```suggestion
Centering an item in a container.
```

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 ---
 
-Placeholder distractor 2
+Vertical alignment of items in a row 
```
```suggestion
Vertical alignment of items in a row.
```

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 ---
 
-Placeholder distractor 3
+Dynamically adjusting distribution and alignment of items for different viewports
```
```suggestion
Dynamically adjusting distribution and alignment of items for different viewports.
```

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --answer--
 
-Placeholder answer
+Fixed sizing and space between items like a grid
```
```suggestion
Fixed sizing and space between items like a grid.
```

henrysmile2071 committed:
fix: typos, grammar errors & remove placeholders

henrysmile2071 committed:
fix: add full stops to sentences

ilenia-magoni commented:
please make a batch from the Edited files tab, do not accept each suggestion as a commit individually

henrysmile2071 commented:
Oh, sorry I wasn't familar with the batch feature, I'll use git squash then.

henrysmile2071 committed:
Apply suggestions from code review

fix: add additional new line between description & remove bracket for element

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

moT01 commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,514 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --text--
 
-Placeholder question
+Which CSS matches the following description?
+
+A container where elements flow from left to right, changes line when it exceeds containter width and new lines go on top of the previous line.
```
```suggestion
What CSS properties would you use to make a `.container` element's items flow from left to right, change lines when an item exceeds container width, and have new items go on top of the previous line?
```

moT01 commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,514 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --text--
 
-Placeholder question
+Which CSS matches the following description?
+
+Center an element within a container horizontally and vertically.
```
```suggestion
What CSS properties would you use to center an element within a flex container horizontally, and vertically?
```

moT01 commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,514 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --text--
 
-Placeholder question
+Items are evenly distributed along the main axis. The spacing between each pair of adjacent items is the same. The items on the sides are flush againt the main axis edge. 
+Which value in `justify-content` matches this description?
```
```suggestion
Which value for `justify-content` will evenly distribute items along the main axis, have the same spacing between each pair of adjacent items, and have items whose sides are flush againt the main axis edge?
```

moT01 commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,514 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --text--
 
-Placeholder question
+Items are evenly distributed along the main axis. The spacing between each pair of adjacent items and against the main axis edge are the same. 
+Which value in `justify-content` matches this description?
```
```suggestion
Which value for `justify-content` will evenly distribute items along the main axis, and have the same spacing between each pair of adjacent items and against the main axis edge?
```

moT01 commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,514 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --text--
 
-Placeholder question
+Items are evenly distributed along the main axis. The spacing between each pair of adjacent items is the same. The items on the sides have half the spacing against the main axis edge. 
+Which value in `justify-content` matches this description?
```
```suggestion
Which value for `justify-content` will evenly distribute items along the main axis, have the same spacing between each pair of adjacent items, and have items on the sides with half the spacing against the main axis edge?
```

moT01 commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,514 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --text--
 
-Placeholder question
+Items are flush against the cross-axis start.
+
+Which value in `align-items` matches this description?
```
```suggestion
Which value for `align-items` will make it so items are flush against the cross-axis start?
```

moT01 commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,514 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --text--
 
-Placeholder question
+Items are flush against the cross-axis end.
+
+Which value in `align-items` matches this description?
```
```suggestion
Which value for `align-items` will make it so items are flush against the cross-axis end?
```

moT01 commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,514 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --text--
 
-Placeholder question
+Items are aligned along the main axis from left to right.
+
+Which value in `flex-direction` matches this description?
```
```suggestion
Which value for `flex-direction` will make it so items are aligned along the main axis from left to right.
```

moT01 commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,514 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --text--
 
-Placeholder question
+Items are aligned along the cross axis from top to bottom. 
+Which value in `flex-direction` matches this description?
```
```suggestion
Which value for `flex-direction` will make it so items are aligned along the cross axis from top to bottom.
```

moT01 reviewed (commented):
This looks good @henrysmile2071 🎉 I like these questions. I would be fine with this as-is, but I made some suggestions to turn some of the questions into single sentences. It's really just a personal preference. I think they're a little more concise and look a little better. Feel free to let me know if you have any issues with any of them, or just want to leave them as they are - some of them do get a little lengthy.

henrysmile2071 commented:
> This looks good @henrysmile2071 🎉 I like these questions. I would be fine with this as-is, but I made some suggestions to turn some of the questions into single sentences. It's really just a personal preference. I think they're a little more concise and look a little better. Feel free to let me know if you have any issues with any of them, or just want to leave them as they are - some of them do get a little lengthy.

@moT01 Thanks, they look much better😁 I will change them as you suggested.

henrysmile2071 committed:
Apply suggestions from code review

Modify descriptive question's structure from multi-sentence to single sentence

Co-authored-by: Tom <20648924+moT01@users.noreply.github.com>

moT01 reviewed (approved):
LGTM @henrysmile2071 🎉 Thanks for making those adjustments 👍

Dario-DC commented on curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:
```diff
@@ -17,439 +17,500 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --text--
 
-Placeholder question
+What does the `stetch` value in `align-items` do?
```
```suggestion
What does the `stretch` value in `align-items` do?
```

Dario-DC reviewed (commented):
Found a typo. Other than that it looks good to me.

jdwilkin4 committed:
Update curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md

Co-authored-by: Dario-DC <105294544+Dario-DC@users.noreply.github.com>

Dario-DC reviewed (approved):
Hi @henrysmile2071 . Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝
